### PR TITLE
Movement code refactor

### PR DESCRIPTION
**Pull request by joaoburatto**
## Summary

Refactors the character movement code to be more straight-forward, concise and cleaner.

## Pictures/Videos (optional)

https://cdn.discordapp.com/attachments/879953211103014972/944483749633085491/6AgUsPQgM6.mp4

## Changes to Files

Added new movement classes, without deleting the old ones.

## Technical Notes (optional)

Adds the animation rigging package to the project.

## Known issues

Needs balancing, maybe implementing animation curves to the input.

### CODE REVIEW

File: Assets/Content/Systems/Player/HumanoidMovementController.cs [hunk 1]
```diff
@@ -13,53 +13,102 @@ namespace SS3D.Content.Systems.Player
...
+        }
+
+        private void ProcessToggleRun()
+        {
             if (Input.GetButtonDown("Toggle Run") && EventSystem.current.currentSelectedGameObject == null)
-            {
                 isWalking = !isWalking;
+        }
+
+        private void ProcessCharacterMovementRefactor()
```
Comment (Alainx277) [inline Assets/Content/Systems/Player/HumanoidMovementController.cs:75]: Remove the `Refactor` at the end and delete the old method.

File: Assets/Content/Systems/Player/IK/SimpleBodyPartLookAt.cs [hunk 2]
```diff
@@ -80,9 +80,7 @@ public void MoveTarget()
 
     public Vector3 GetMousePosition(bool changeYAxis)
     {
-        Ray ray = camera.ScreenPointToRay(Input.mousePosition);
-        Vector3 mousePos = ray.origin - ray.direction * (ray.origin.y / ray.direction.y);
-        mousePos = new Vector3(mousePos.x, changeYAxis ? mousePos.y : transform.position.y, mousePos.z);
+
```
Comment (Alainx277) [inline Assets/Content/Systems/Player/IK/SimpleBodyPartLookAt.cs:84]: Why have two empty lines at the top of the body?

File: Assets/Content/Systems/Player/Movement/BipedMovementController.cs [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+namespace SS3D.Content.Systems.Player.Movement
+{
+    /// <summary>
+    /// Controls the movement for biped characters that use the same armature
+    /// as the human model uses.
+    /// </summary>
+    [RequireComponent(typeof(BipedAnimatorController))]
+    [RequireComponent(typeof(CharacterController))]
+    [RequireComponent(typeof(Animator))]
+    public class BipedMovementController : NetworkBehaviour
```
Comment (Alainx277) [inline Assets/Content/Systems/Player/Movement/BipedMovementController.cs:16]: What's the relation of this with the `HumanoidMovementController`?

File: Assets/TemporaryCopyBones.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿using UnityEngine;
+
+namespace SS3D
+{
+    public class TemporaryCopyBones : MonoBehaviour
```
Comment (Alainx277) [inline Assets/TemporaryCopyBones.cs:5]: What is the purpose of this class? Maybe you could add a comment.

**Review by Alainx277 — CHANGES_REQUESTED** [review]

File: ProjectSettings/QualitySettings.asset [hunk 5]
```diff
@@ -117,7 +117,7 @@ QualitySettings:
   - serializedVersion: 2
     name: High
     pixelLightCount: 4
-    shadows: 2
+    shadows: 0
```
Comment (cosmiccoincidence) [inline ProjectSettings/QualitySettings.asset:120]: Why change quality settings?

File: Assets/Content/Systems/Player/Movement/BipedMovementController.cs (hunk 3 above)
Comment (joaoburatto) [reply Assets/Content/Systems/Player/Movement/BipedMovementController.cs:16]: I'll discard the HumanoidMovementController class

File: Assets/TemporaryCopyBones.cs (hunk 4 above)
Comment (joaoburatto) [reply Assets/TemporaryCopyBones.cs:5]: Deleted the class

File: ProjectSettings/QualitySettings.asset (hunk 5 above)
Comment (joaoburatto) [reply ProjectSettings/QualitySettings.asset:120]: Will revert

File: Assets/Content/Systems/Player/IK/SimpleBodyPartLookAt.cs [hunk 6]
```diff
@@ -1,89 +1,93 @@
...
+            public bool z;
+        }
 
 #if UNITY_EDITOR
-    // This class is resposable of displaying the bools properly. Values are hardcoded for lack of better means. (Currently unused)
-    //[CustomPropertyDrawer(typeof(Limits))]
-    private class LimitsUIE : PropertyDrawer
-    {
-        public override void OnGUI(Rect position, SerializedProperty property, GUIContent label)
+        // This class is resposable of displaying the bools properly. Values are hardcoded for lack of better means. (Currently unused)
```
Comment (Alainx277) [inline Assets/Content/Systems/Player/IK/SimpleBodyPartLookAt.cs:36]: What does "Currently unused" mean?
```suggestion
        // This class is responsible for displaying the bools properly. Values are hardcoded for lack of better means. (Currently unused)
```

**Review by Alainx277 — COMMENTED** [review]
Looks good, just one minor thing.

**Review by nsarrazin — CHANGES_REQUESTED** [review]
PR looks good, only issue is that the animation rigging package has a dependency on Burst which by default breaks linux compatibility. Updating Burst to the latest availabe version resolves it, though. 

I've suggested the correct version numbers.

**Comment by nsarrazin** [comment]
I tested this functionally and it worked pretty well. I tested it locally and in multiplayer. For me it's looking good, I will review the merge conflict tomorrow and merge this.

**Comment by nsarrazin** [comment]
I tested this functionally and it worked pretty well locally. I updated the weights & input curves to make it more snappy but it looks good. It doesn't work over the network though, I'm investigating it and will suggest changes when I'm done.

**Comment by kiiada** [comment]
> I tested this functionally and it worked pretty well locally. I updated the weights & input curves to make it more snappy but it looks good. It doesn't work over the network though, I'm investigating it and will suggest changes when I'm done.

What issues are you seeing over network? I tried this both with multiple instances running locally and from multiple machines on the same network and movement and animations stayed synced on both.

The only thing that I noticed is that when the user is typing in the chat window the character controller still uses the input for movement

**Comment by nsarrazin** [comment]
> What issues are you seeing over network? I tried this both with multiple instances running locally and from multiple machines on the same network and movement and animations stayed synced on both.

Yeah my bad networking worked just fine, forgot to update here.

**Comment by joaoburatto** [comment]
Closing due to this being merged on develop for #920

**Pull request closed without merging**